content/articles/create-a-java-getter-setter-generator-using-python/index.md

```diff
@@ -0,0 +1,228 @@
+author: Dennis-Kariuki
+date: 2021-11-23T00:00:00-05:35
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/create-a-java-getter-setter-generator-using-python/hero.png
+    alt: Create a Java getter-setter generator using Python Hero image
+
+--- 
+Python's getters and setters are unique. They allow data encapsulation while others have hidden fields for private variables. 
+<!--more-->
+Several OOP languages use getters and setters to store data. We conceal an object class's attributes so other classes' methods do not alter data accidentally.
+
+In object-oriented languages, these methods get and change data. Setting an object's attribute value is called `setting`. In this article, we will learn how to use Python's getter and setter methods.
+
+### Table of contents
+- [Understanding getters and setters](#understanding-getters-and-setters)
+- [Get getters and setters behavior using normal functions](#get-getters-and-setters-behavior-using-normal-functions)
+- [Get getters and setters behavior using @Property decorators](#get-getters-and-setters-behavior-using-@Property-decorators)
+- [Get getters and setters behavior using the property() function](#get-getters-and-setters-behavior-using-the-roperty()-function)
+- [Project algorithm](#project-algorithm)
+- [Implementation of the algorithms](#implementation-of-the-algorithms)
+- [Conclusion](#conclusion)
+
+### Understanding getters and setters
+The value which a getter method returns is its argument. This is useful for OOP private property access. A setter modifies a property's value. 
+
+This helps establish private class characteristics in object-oriented programming. Using getters and setters ensures there is data encapsulation.
+
+Why we use setters and getters:
+- To promote encapsulation.
+- To keep the interface even if the internals change.
+- To add validation logic to value acquisition and setting.
+- To externalize the property's internal representation.
+- To ensure there is no direct access or change to a class field.
+- Access levels are configured using getters and setters.
+
+### Get getters and setters behavior using normal functions
+If we use the standard `get()` and `set()` methods to get the getters and setters property, there will be no need for any extra specific implementation code.
+
+```python 
+class section:
+    def __init__(ascetic, span = 0):   
+         ascetic._span = span     
+    def get_span(ascetic):   
+        return ascetic._span    
+    def set_span(ascetic, k):   
+        ascetic._span = k  
+    
+kelvin = section()   
+kelvin.set_span(34)   
+print(kelvin.get_span())   
+print(kelvin._span)
+```
+
+Output:
+
+```bash 
+34
+34
+```
+
+Because the methods `get age()` and `set age()` are regular functions, they may provide the needed functionality. Python has a distinguishing function ().

```

Comment:
Expound on the last sentence here and explain what is happening in your code better